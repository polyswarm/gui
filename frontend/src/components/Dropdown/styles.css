.Dropdown {
    background-color: rgb(46, 255, 109);
    border-radius: 4px;
    box-shadow: inset 0 -4px rgba(0, 0, 0, .1) ;
    display: inline-block;
    bottom: 0;
    height: 48px;
    margin-top: auto;
    margin-right: 8px;
    min-width: 130px;
}

.Dropdown:after {
    border-top: 10px white solid;
    border-left: 5px transparent solid;
    border-right: 5px transparent solid;
    content: '';
    right: -85%;
    bottom: 18px;
    height: 0;
    position: relative;
    width: 0;
}

.Dropdown:hover {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    box-shadow: none;
}

.Dropdown > p {
    font-size: 20px;
    height: 20px;
    line-height: 20px;
    padding-left: 8px;
    margin: 14px 0;
}

.Dropdown-Choices {
    display: none;
}

.Dropdown:hover > .Dropdown-Choices {
    background-color: #dedede;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
    display: block;
    position: relative;
    overflow: hidden;
    width: 130px;
    z-index: 1;
}

.Dropdown-Choices > p {
    border-bottom: black;
    color: black;
    margin: 0;
    padding: 8px;
}

.Dropdown-Choices > p:hover {
    background-color: #a64ef8;
    color: white;
}

.Dropdown-Choices > p:active {
    background-color: #9125f7;
    color: white;
}