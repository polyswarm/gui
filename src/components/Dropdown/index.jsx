// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Project imports
// Component imports
class Dropdown extends Component {
  render() {
    const {props: {children}} = this;
    return(
      <div className='Dropdown'>
        <img className='Dropdown-Icon'
          src='../public/img/menu.svg'
          width='7px'
          height='24px'
          alt='menu'/>
        <div className='Dropdown-Choices'>
            {children}
        </div>
      </div>
    );
  }
}
Dropdown.proptypes = {
}
export default Dropdown;