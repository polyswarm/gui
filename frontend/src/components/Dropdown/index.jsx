// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Project imports
// Component imports
import './styles.css';
import strings from './strings.js';

class DirectionDropdown extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selected: 0
    }

    this.onSelectionChanged = this.onSelectionChanged.bind(this);
  }

  render() {
    const {state: {selected}} = this;
    return(
      <div className='Dropdown'>
        <p className='Dropdown-Selection'>
            {selected == 0 ? strings.deposit : strings.withdraw}
        </p>
        <div className='Dropdown-Choices'>
            <p onClick={() => this.onSelectionChanged(0)}>
            {strings.deposit}
            </p>
            <p onClick={() => this.onSelectionChanged(1)}>
            {strings.withdraw}
            </p>
        </div>
      </div>
    );
  }

  onSelectionChanged(index) {
    const {props: {onSelectionChanged}, state: {selected}} = this;
    if (index == selected) {
        return;
    }

    if (onSelectionChanged) {
        onSelectionChanged(index);
    }
    this.setState({selected: index});
  }

}
DirectionDropdown.proptypes = {
  onSelectionChanged: PropTypes.func
}
export default DirectionDropdown;