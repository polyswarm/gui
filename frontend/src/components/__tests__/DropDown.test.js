import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Dropdown from '../Dropdown';

it('renders without crashing', () => {
  const wrapper = render(<Dropdown />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('Shows Deposit and Withdraw when hovered over main part', () => {
  const wrapper = mount(<Dropdown />);

  wrapper.simulate('mouseEnter');
  
  expect(wrapper.find('.Dropdown-Choices').find('p').first().text()).toEqual('Deposit');
  expect(wrapper.find('.Dropdown-Choices').find('p').last().text()).toEqual('Withdraw');
});

it('Changes selected value when an option is clicked', () => {
  const wrapper = mount(<Dropdown />);
  wrapper.simulate('mouseEnter');

  wrapper.find('.Dropdown-Choices').find('p').last().simulate('click');
  
  expect(wrapper.find('p').first().text()).toEqual('Withdraw');

});

it('Calls setState when a row is clicked', () => {
  const setState = jest.spyOn(Dropdown.prototype, 'setState');
  const wrapper = mount(<Dropdown />);

  wrapper.simulate('mouseEnter');
  wrapper.find('.Dropdown-Choices').find('p').last().simulate('click');


  expect(setState).toHaveBeenCalledWith({selected: 1});
});

it('Calls onSelectionChanged when a row is clicked', () => {
  const onSelectionChanged = jest.fn();
  const wrapper = mount(<Dropdown onSelectionChanged={onSelectionChanged} />);

  wrapper.simulate('mouseEnter');
  wrapper.find('.Dropdown-Choices').find('p').last().simulate('click');


  expect(onSelectionChanged).toHaveBeenCalledWith(1);
});

it('Does not call onSelectionChanged when the already selected row is clicked', () => {
  const onSelectionChanged = jest.fn();
  const wrapper = mount(<Dropdown onSelectionChanged={onSelectionChanged} />);

  wrapper.simulate('mouseEnter');
  wrapper.find('.Dropdown-Choices').find('p').first().simulate('click');


  expect(onSelectionChanged).toHaveBeenCalledTimes(0);
});