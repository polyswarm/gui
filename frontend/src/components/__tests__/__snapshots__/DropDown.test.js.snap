// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Dropdown"
>
  <p
    class="Dropdown-Selection"
  >
    Deposit
  </p>
  <div
    class="Dropdown-Choices"
  >
    <p>
      Deposit
    </p>
    <p>
      Withdraw
    </p>
  </div>
</div>
`;
